packages/contractkit/src/wrappers/Election.ts

```diff
@@ -475,9 +472,13 @@ export class ElectionWrapper extends BaseWrapper<Election> {
    * @param address The address to retrieve VoterRewards for.
    * @param epochNumber The epoch to retrieve VoterRewards at.
    */
-  async getVoterRewards(address: Address, epochNumber: number): Promise<VoterReward[]> {
+  async getVoterRewards(
+    address: Address,
+    epochNumber: number,
+    archive?: boolean
+  ): Promise<VoterReward[]> {
     const blockNumber = await this.kit.getLastBlockNumberForEpoch(epochNumber)
-    const voter = await this.getVoter(address, blockNumber)
+    const voter = await this.getVoter(address, archive ? blockNumber : undefined)
```

Comment:
This implementation is misleading because if the user passes in a past epoch and `archive=false` it will give them incorrect information. I don't think we should add the archive option. Instead the epochNumber option should be optional and if specified will attempt, but possibly fail, to retrieve past data.

Reply:
Thanks. Very good points. I've updated to call `estimate` and default false. Without an archive node (or employing approximation) there's only 85 minute (1024 * 5 seconds) window to retrieve exact voter rewards information.